docs/interfaces/keyring.md

```diff
@@ -0,0 +1,87 @@
+<!--
+order: 3
+synopsis: "This document describes how to configure and use the keyring and its various backends for an [**application**](../basics/app-anatomy.md). A separate document for implementing a CLI for an SDK [**module**](../building-modules/intro.md) can be found [here](#../building-modules/module-interfaces.md#cli)."
+-->
+
+# The keyring
+
+Starting with the v0.38.0 release, Cosmos SDK comes with a new keyring implementation
+that provides a set of commands to manage cryptographic keys in a secure fashion. The
+new keyring supports multiple storage backends, some of which may not be available on
+all operating systems.
+
+## The 'os' backend
+
+The `os` backend relies on operating system-specific defaults to handle key storage
+securely. Typically, operating systems credentials sub-systems handle passwords prompt,
+private keys storage, and user sessions according to their users password policies. Here
+is a list of the most popular operating systems and their respective passwords manager:
+
+* macOS (since Mac OS 8.6): [Keychain](https://support.apple.com/en-gb/guide/keychain-access/welcome/mac)
+* Windows: [Credentials Management API](https://docs.microsoft.com/en-us/windows/win32/secauthn/credentials-management)
+* GNU/Linux:
+  * [libsecret](https://gitlab.gnome.org/GNOME/libsecret)
+  * [kwallet](https://api.kde.org/frameworks/kwallet/html/index.html)
+
+GNU/Linux distributions that use GNOME as default desktop environment typically come with
+[Seahorse](https://wiki.gnome.org/Apps/Seahorse). Users of KDE based distributions are
+commonly provided with [KDE Wallet Manager](https://userbase.kde.org/KDE_Wallet_Manager).
+Whilst the former is in fact a `libsecret` convenient frontend, the former is a `kwallet`
+client.
+
+`os` is the default option since operating system's default credentials managers are
+designed to meet users' most common needs and provide them with a comfortable
+experience without compromising on security.
+
+## The 'file' backend
+
+The `file` backend more closely resembles the keybase implementation used prior to
+v0.38.1. It stores the keyring encrypted within the apps configuration directory. This
+keyring will request a password each time it is accessed, which may occur multiple
+times in a single command resulting in repeated password prompts. If using bash scripts
+to execute commands using the `file` option you may want to utilize the following format
+for multiple prompts (assuming that keyring's password in stored in the `$KEYPASSWD`
+environment variable):
+
+```sh
+$ gaiacli config keyring-backend file         # use file backend
+$ echo "${KEYPASSWD}" | gaiacli keys add me   # add the key 'me'
+$ echo -e "${KEYPASSWD}\n${KEYPASSWD}\n${KEYPASSWD}" | gaiad collect-gentxs # multiple prompts
```

Comment:
I had trouble with this format, it never worked for me. I ended up using this and it worked:
```
(echo '1234567890'; echo '1234567890') | ebcli keys add testuser
```
Maybe it would be good to list both methods?

Reply:
https://github.com/cosmos/cosmos-sdk/pull/5854/commits/3cef329228b31abfa25e37e9d3b59adfd780f2ae